karman-avionics/karman-avionics/src/framework/Timer.c

```diff
@@ -24,5 +24,19 @@ void timer_init(void){
 }
 
 inline uint32_t get_timer_count(void){
-    return timerCount;
+    irqflags_t flags = cpu_irq_save();
+    uint32_t count = timerCount;
+    cpu_irq_restore(flags);
+    return count;
 }
+
+inline void timer_delay_ms(uint8_t millis)
+{
+    uint32_t timer_begin = get_timer_count();
+    uint32_t timer_end = timer_begin + 2*millis;
+    while(get_timer_count() != timer_end)
```

Comment:
It's probably safer to use `get_timer_count() < timer_end` just in case we miss the exact endpoint.